source/extensions/common/wasm/context.cc

```diff
@@ -202,6 +204,53 @@ void Context::onResolveDns(uint32_t token, Envoy::Network::DnsResolver::Resoluti
   wasm()->on_resolve_dns_(this, id_, token, s);
 }
 
+void Context::onStat(Envoy::Stats::MetricSnapshot& snapshot) {
+  // {name}.{delta/value}|{c/g}/n
+  proxy_wasm::DeferAfterCallActions actions(this);
+  uint32_t s = 0;
+  for (const auto& counter : snapshot.counters()) {
+    if (counter.counter_.get().used()) {
+      s += absl::StrCat(counter.counter_.get().name(), ".", counter.delta_, "|c/n").size() + 1;
```

Comment:
You might want to use a different separator.  "." is a common selement and separator in the counter names.  Try something which isn't used e.g. \0 or maybe space.

Reply:
See if you can calculate the size without actually constructing the string (for efficiency).

Reply:
switched to use a space as a separator and calculated size without string creation